### PR TITLE
CRUD Structs

### PR DESCRIPTION
**Pull request by TNTzx**
Adds a few CRUD controls for `musicLabel`s and `artist`s:
- Create
- Get by ID or name
- Update
- Delete

This PR also includes new `UpdateTemplate`s.

### CODE REVIEW

File: src/structures/artist.ts [hunk 1]
```diff
@@ -34,3 +35,108 @@ export function artistCTToInput(artistCT: ArtistCreateTemplate): Prisma.ArtistCr
...
+
+    /**
+     * Returns all artists with the name using an exact search.
+     * 
+     * @param name The artist name.
+     * @returns The artists with that name.
+     */
+    static async getFromName(name: string) {
+        return await Database.artist.findMany({
+            where: { name: name },
```
Comment (naoei) [inline src/structures/artist.ts:126]: This should get from `safe_name` instead. which hasn't been implemented in the schema, so you'll have to do that as well - and by extension, generate the `safe_name` in `create()`.

Same applies to music labels.

**Review by naoei — CHANGES_REQUESTED** [review]
I don't know if I will use most of these functions within the routers since these CRUD structs are mainly just shorthands for the longer and direct Prisma approach. It could be warranted if there were some extra logic for these.

**Comment by TNTzx** [comment]
sometimes some CRUD operations could require an update (like some security vuln fix or extra validation / conversion on the template) and it would be bad to copy paste that update on all endpoints that need to do anything related to the operation being used

like for example the safe name part, thought that could be put in `artistCTToInput` instead

plus it's easier to put things into a function since that abstracts it so that you won't have to type out `Database.artist.create(...)`

File: src/structures/artist.ts (hunk 1 above)
Comment (TNTzx) [reply src/structures/artist.ts:126]: i'll open a new PR to include `safeName` in both `artist`s and `musicLabel`s

**Comment by naoei** [comment]
I guess it's fine then.

**Review by naoei — APPROVED** [review]

**Pull request merged**